File: LeetCode/Study Plan/Practice Problems/Easy/Shortest Word Distance/Go/shortest_word_distance.go
package main

import (
	"fmt"
	"math"
)

func shortest_word_distance(words_dict []string, word1, word2 string) int {
	ind1, ind2 := -1, -1
	dis := math.MaxInt64
	n := len(words_dict)
	for i := 0; i < n; i++ {
		if words_dict[i] == word1 {
			ind1 = i
		}
		if words_dict[i] == word2 {
			ind2 = i
		}
		if ind1 != -1 && ind2 != -1 {
			dis = min(dis, abs(ind1-ind2))
		}
	}
	return dis
}

func abs(a int) int {
	if a > 0 {
		return a
	}
	return -1 * a
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

func main() {
	fmt.Println(shortest_word_distance([]string{"practice", "makes", "perfect", "coding", "makes"}, "coding", "practice"))
}

File: LeetCode/Easy/Shortest Word Distance/Cpp/shortest_word_distance.cc
#include <iostream>
#include <vector>

typedef std::vector<std::string> vs;

int shortest_word_distance(vs words_dict, std::string word1, std::string word2) {
    int ind1 = -1, ind2 = -1;
    int dis = INT_MAX;
    for (int i=0; i<words_dict.size(); i++) {
        if (words_dict[i].compare(word1) == 0) ind1 = i;
        if (words_dict[i].compare(word2) == 0) ind2 = i;
        if (ind1 != -1 && ind2 != -1) dis = std::min(dis, abs(ind1-ind2));
    }
    return dis;
}

int main() {
    std::cout<<shortest_word_distance(vs{"practice", "makes", "perfect", "coding", "makes"}, "practice", "coding")<<"\n";
}
